src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -108,14 +105,11 @@ private DecisionTree AddByValue(DecisionTree.Guarded guarded, BoundConstantPatte
         {
             if (guarded.Default != null)
             {
-                if (guarded.Default.MatchIsComplete)
-                {
-                    return null;
-                }
+                Debug.Assert(!guarded.Default.MatchIsComplete); // otherwise we would have given a subsumption error
```

Comment:
>Debug.Assert(!guarded.Default.MatchIsComplete); // otherwise we would have given a subsumption error [](start = 16, length = 100)

Should we throw instead?